### PR TITLE
[INLONG-6042][Manager] Support delete and update DataNode by unique key

### LINKED ISSUE #6042: [Feature][Manager] Support delete and update datanode by unique key
### Description

_No response_

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vernedeng**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #6042 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/node/DataNodeServiceImpl.java [hunk 1]
```diff
@@ -142,6 +142,20 @@ public Boolean delete(Integer id, String operator) {
             return false;
         }
 
+        return delete(entity, operator);
+    }
+
+    @Override
+    public Boolean deleteByName(String name, String operator) {
+        DataNodeEntity entity = dataNodeMapper.selectByNameAndType(name, null);
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/node/DataNodeServiceImpl.java:150]: There may return more than one entity.
Suggested make tye type not null.

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/client/DataNodeClient.java [hunk 2]
```diff
@@ -107,4 +107,17 @@ public Boolean delete(Integer id) {
         return response.getData();
     }
 
+    /**
+     * Delete data node by name.
+     *
+     * @param name node name to be deleted
+     * @return whether succeed
+     */
+    public Boolean delete(String name) {
```
Comment (healchow) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/client/DataNodeClient.java:116]: Please specify the data node type at the same time.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/node/DataNodeServiceImpl.java (hunk 1 above)
Comment (vernedeng) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/node/DataNodeServiceImpl.java]: fixed, thx

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/client/DataNodeClient.java (hunk 2 above)
Comment (vernedeng) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/inner/client/DataNodeClient.java]: fixed, thx

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/node/custom/CustomDataNodeOperator.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import org.apache.inlong.manager.service.node.AbstractDataNodeOperator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+/**
+ * Custom data node operator
+ */
+@Service
+public class CustomDataNodeOperator extends AbstractDataNodeOperator {
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/node/custom/CustomDataNodeOperator.java:37]: What is the relation between the `delete data node` and `custom data node`?

**Review by luchunliang — APPROVED** [review]
LGTM

**Review by gosonzhang — APPROVED** [review]

**Review by healchow — APPROVED** [review]

**Pull request merged**